server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -668,10 +669,26 @@ void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, String pipelin
         ExceptionsHelper.rethrowAndSuppress(exceptions);
     }
 
+    private record IngestPipelinesExecutionResult(boolean success, boolean kept, Exception exception, String failedIndex) {}
+
+    private static IngestPipelinesExecutionResult successResult() {
+        return new IngestPipelinesExecutionResult(true, true, null, null);
+    }
+
+    private static IngestPipelinesExecutionResult discardResult() {
+        return new IngestPipelinesExecutionResult(true, false, null, null);
+    }
```

Comment:
Could these two could be static constants, since nothing is being changed? Something like:

```java
private static IngestPipelinesExecutionResult SUCCESS_RESULT = new IngestPipelinesExceptionResult(true, true, null, null);
```

Then we could avoid creating a new object each time.